Dockerfile

```diff
@@ -18,6 +19,8 @@ CMD ["node", "index.js"]
 # Production
 FROM node:${NODE_VERSION}-alpine AS production
 
+ARG NODE_ENV
```

Comment:
This argument is declared already at the top of the Dockerfile

Reply:
ARG's declared outside the FROM block are not seen inside the FROM block, unless a blank ARG is seen inside the FROM block. This has resulted in the ENV NODE_ENV $NODE_ENV line setting NODE_ENV to a blank value, which causes the server to fail at runtime.

Reply:
Ah, my mistake. In that case, we should move the `ARG NODE_ENV=production` from the top of the Dockerfile to the `production` stage, rather than leaving an unused ARG at the top. Keep the default value of `production` so that we can't accidentally run a development build by forgetting to add the argument in the CI pipeline.